pkg/member/member_control.go

```diff
@@ -299,3 +303,30 @@ func (m *memberControl) IsLearnerPresent(ctx context.Context) (bool, error) {
 
 	return miscellaneous.CheckIfLearnerPresent(learnerCtx, cli)
 }
+
+// IsScaleup determines whether a etcd cluster is getting scale-up or not.
+func IsScaleup(ctx context.Context, m Control) (bool, error) {
```

Comment:
Can we do the following:

1. Rename this method to `clusterScaled`
2. Make it a method in `Control` interface instead of passing an instance of `Control` to it.
3. This method does 2 completely different things which do not belong in a single method:
    - It checks if the cluster is scaled from nonHA to HA by calling `miscellaneous.GetInitialClusterStateIfScaleup`
    - It check if the current member is already in cluster.

Reply:
> This method does 2 completely different things which do not belong in a single method:
It checks if the cluster is scaled from nonHA to HA by calling miscellaneous.GetInitialClusterStateIfScaleup
It check if the current member is already in cluster.

What makes you think that ?
IMO it `IsScaleup` does only 1 thing which it checks cluster is getting scaled-up or not using these 2 function calls:
`miscellaneous.GetInitialClusterStateIfScaleup` and `IsMemberInCluster`.

If `miscellaneous.GetInitialClusterStateIfScaleup`  says I have the annotation in sts --> surely it is scaleup case return true no need to call `IsMemberInCluster`

If `miscellaneous.GetInitialClusterStateIfScaleup` says I don't have annotation in sts  --> we still not sure about it is not being scaleup case due to bug in adding annotation to sts: https://github.com/gardener/etcd-druid/issues/557

Reply:
>  Rename this method to clusterScaled

what about `IsClusterScaledup` ?

Reply:
>what about IsClusterScaledup ?

Maybe we can use `IsClusterScaledUp`